pyDeltaRCM/model.py

```diff
@@ -114,6 +119,12 @@ def finalize(self):
 
         """
         self.logger.info('Finalize model run')
+
+        # get the final timestep recorded, if needed.
+        if self._save_time_since_last >= self.save_dt:
+            self.record_stratigraphy()
+            self.output_data()
```

Comment:
Do you think there is any merit to **always** saving when `finalize()` is called? (dropping the `if` conditional)

Yes this would potentially throw off a regular save interval if the run is not due for a save. But, this would ensure that the 'final' state of the model is captured (might be nice in the case of a run with `save_dt = 150`, `dt = 15` and a final time of `295` as an example?).  Otherwise a run could just go on and run many iterations, depending on the configuration, that are ultimately not saved.

Alternatively I could see the benefits of outputting a warning to the user telling them that `X` number of timesteps or amount of time will be run but not saved. Or using the `save_dt` and `dt` information to cut a run short and avoid wasted iterations... These concerns are not really pertinent to this PR, just something that occurred to me.

Reply:
It's a fair question, and one that I asked myself too. I felt that adding the `if` was the least offensive of the options, because it just kind of does exactly what you expect, without anything fancy.

That said, I agree with you, it probably makes sense to save the final state *always*. Someone could always drop it in post-processing if they wanted to.

I'm going to open an issue with this concern, rather than changing it now.